src/window/window_wrapper.rs

```diff
@@ -509,10 +509,12 @@ impl WinitWindowWrapper {
         // It's important that this is created before the window is resized, since it can change the padding and affect the size
         #[cfg(target_os = "macos")]
         {
-            self.macos_feature = Some(MacosWindowFeature::from_winit_window(
-                window,
-                self.settings.clone(),
-            ));
+            let feature = MacosWindowFeature::from_winit_window(window, self.settings.clone());
+            WINDOW_FEATURE.with(|cell| {
+                cell.set(feature.clone())
+                    .expect("WINDOW_FEATURE already set");
+            });
```

Comment:
Can't you call `window_feature.ns_window.makeKeyAndOrderFront(None);` directly here, instead of using storing it in a thread local variable? Additionally it's already stored in `self.macos_feature`, so you should be able to read it from there.